pkg/controllers/user/nodesyncer/nodetaints.go

```diff
@@ -41,14 +43,21 @@ func (n *nodesSyncer) syncTaints(key string, obj *v3.Node) (runtime.Object, erro
 			taintList = append(taintList, taintStr)
 		}
 		newNode.Spec.Taints = taintList
-		if _, err := n.nodeClient.Update(newNode); err != nil {
+		if _, err := n.nodeClient.Update(newNode); err != nil && !isDuplicate(err) {
 			return obj, errors.Wrapf(err, "failed to update corev1.Node %s from v3.Node %s in node taint controller", node.Name, obj.Name)
-		}
-		if !reflect.DeepEqual(newObj.Spec.DesiredNodeTaints, newObj.Spec.InternalNodeSpec.Taints) {
+		} else if isDuplicate(err) {
```

Comment:
I think this will error out even if there's one duplicate, do we want to set desired taints to nil in this case? 
Is there any way we can calculate taintList so that this conflict doesn't happen?

Reply:
The point for this duplicate check is that, in k8s >=1.14, they make key and effect as unique but only key in version <=1.13. And I don't want to block the following sync if this error happens.

Reply:
we should add api validation in store update similar to what UI does so it's clearly visible to user

Reply:
Updated